rdmo/projects/viewsets.py

```diff
@@ -778,19 +838,64 @@ class ValueViewSet(ReadOnlyModelViewSet):
     permission_classes = (HasModelPermission | HasProjectsPermission, )
     serializer_class = ValueSerializer
 
-    filter_backends = (SnapshotFilterBackend, DjangoFilterBackend)
+    filter_backends = (
+        AttributeFilterBackend,
+        SnapshotFilterBackend,
+        OptionFilterBackend,
+        DjangoFilterBackend,
+        SearchFilter
+    )
     filterset_fields = (
         'project',
         # snapshot is part of SnapshotFilterBackend
-        'attribute',
+        # attribute is part of AttributeFilterBackend
         'attribute__uri',
-        'option',
+        'attribute__path',
+        # option is part of OptionFilterBackend
         'option__uri',
+        'option__uri_path',
     )
 
+    search_fields = ['text', 'project__title', 'snapshot__title']
+
     def get_queryset(self):
         return Value.objects.filter_user(self.request.user).select_related('attribute', 'option')
 
+    @action(detail=False, permission_classes=(HasModelPermission | HasProjectsPermission, ))
+    def search(self, request):
+        queryset = self.filter_queryset(self.get_queryset()).exclude_empty().select_related('project', 'snapshot')
+
+        # add a subquery to get the label for the set the value is part in
+        try:
+            set_attribute = int(request.GET.get('set_attribute'))
+            set_label_subquery = Subquery(
+                Value.objects.filter(attribute=set_attribute, set_prefix='', set_index=OuterRef('set_index'))
+                             .values('text')[:1]
+            )
+            queryset = queryset.annotate(set_label=set_label_subquery)
+        except (ValueError, TypeError):
+            pass
+
+        if is_truthy(request.GET.get('collection')):
+            # if collection is set (for checkboxes), we first select each distinct set and create a Q object with it
+            # by doing so we can select an undetermined number of values which belong to an exact number of sets
+            # given by settings.PROJECT_VALUES_SEARCH_LIMIT
+            values_list = queryset.order_by('project_id', 'snapshot_id', 'attribute_id', 'set_prefix', 'set_index') \
+                                  .distinct('project_id', 'snapshot_id', 'attribute_id', 'set_prefix', 'set_index') \
```

Comment:
nice query, but unfortunately `DISTINCT ON` is not supported by sqlite (https://code.djangoproject.com/ticket/22696). There goes everybody's' local dev setup 🙃 
Want to have a backend specific work-around query for sqlite maybe?